bin/fuel-core/src/cli/snapshot.rs

```diff
@@ -44,55 +101,137 @@ pub enum SubCommands {
+    let stdout = std::io::stdout().lock();
+    serde_json::to_writer_pretty(stdout, &config)
+        .context("failed to dump contract snapshot to JSON")?;
+    Ok(())
+}
+
+fn full_snapshot(
+    prev_chain_config: Option<PathBuf>,
+    output_dir: &Path,
+    encoding: Encoding,
+    db: impl ChainStateDb,
+) -> Result<(), anyhow::Error> {
+    std::fs::create_dir_all(output_dir)?;
+
+    let metadata = write_metadata(output_dir, encoding)?;
+
+    write_chain_state(&db, &metadata)?;
+    write_chain_config(db, prev_chain_config, &metadata.chain_config())?;
+    Ok(())
+}
+
+fn write_chain_config(
+    db: impl ChainStateDb,
+    chain_config: Option<PathBuf>,
+    file: &Path,
+) -> Result<(), anyhow::Error> {
+    let height = db.get_block_height()?;
+    let chain_config = ChainConfig {
+        height: Some(height),
+        ..load_chain_config(chain_config)?
+    };
+
+    chain_config.create_config_file(file)
+}
+
+fn write_metadata(dir: &Path, encoding: Encoding) -> anyhow::Result<SnapshotMetadata> {
+    match encoding {
+        Encoding::Json => SnapshotMetadata::write_json(dir),
+        #[cfg(feature = "parquet")]
+        Encoding::Parquet {
+            compression,
+            group_size,
+            ..
+        } => SnapshotMetadata::write_parquet(dir, compression.try_into()?, group_size),
+    }
+}
+
+fn write_chain_state(
+    db: impl ChainStateDb,
+    metadata: &SnapshotMetadata,
+) -> anyhow::Result<()> {
+    let mut encoder = Encoder::for_snapshot(&metadata)?;
+    fn write<T>(
+        data: impl Iterator<Item = StorageResult<T>>,
+        group_size: usize,
+        mut write: impl FnMut(Vec<T>) -> anyhow::Result<()>,
+    ) -> anyhow::Result<()> {
+        data.chunks(group_size)
+            .into_iter()
+            .try_for_each(|chunk| write(chunk.try_collect()?))
     }
+    let group_size = metadata
+        .state_encoding()
+        .group_size()
+        .unwrap_or(MAX_GROUP_SIZE);
+
+    let coins = db.iter_coin_configs();
+    write(coins, group_size, |chunk| encoder.write_coins(chunk))?;
+
+    let messages = db.iter_message_configs();
+    write(messages, group_size, |chunk| encoder.write_messages(chunk))?;
+
+    let contracts = db.iter_contract_configs();
+    write(contracts, group_size, |chunk| {
+        encoder.write_contracts(chunk)
+    })?;
+
+    let contract_states = db.iter_contract_state_configs();
+    write(contract_states, group_size, |chunk| {
+        encoder.write_contract_state(chunk)
+    })?;
+
+    let contract_balances = db.iter_contract_balance_configs();
+    write(contract_balances, group_size, |chunk| {
+        encoder.write_contract_balance(chunk)
+    })?;
+
+    encoder.close()?;
+
     Ok(())
 }
+
+fn load_chain_config(
+    chain_config: Option<PathBuf>,
+) -> Result<ChainConfig, anyhow::Error> {
+    let chain_config = match chain_config {
+        Some(file) => ChainConfig::load(file)?,
+        None => ChainConfig::local_testnet(),
```

Comment:
Idk if I like this implicit behavior. This method should only do the loading. The caller should handle mapping a `None`. 

I also think there should be a message emitted or some way to notify the caller that they are getting a `local_testnet` config.